app/scripts/metamask-controller.js

```diff
@@ -5566,20 +5574,11 @@ export default class MetamaskController extends EventEmitter {
       delete permissions[PermissionNames.permittedChains];
     }
 
-    const id = nanoid();
-    const legacyApproval =
-      await this.approvalController.addAndShowApprovalRequest({
-        id,
-        origin,
-        requestData: {
-          metadata: {
-            id,
-            origin,
-          },
-          permissions,
-        },
-        type: MethodNames.RequestPermissions,
-      });
+    const requestedChains =
+      permissions[PermissionNames.permittedChains]?.caveats?.[0]?.value ?? [];
+
+    const requestedAccounts =
+      permissions[PermissionNames.eth_accounts]?.caveats?.[0]?.value ?? [];
```

Comment:
we really should be making sure the caveat has the right type. i.e. `restrictReturnedAccounts` or `restrictReturnedNetworks`. It probably is fine to assume the first caveat is the right caveat, but it feels wrong to assume that 😅